sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -610,13 +610,17 @@ trait ParquetPartitionReaderBase extends Logging with Arm with ScanWithMetrics
       inputTable: Table,
       filePath: String,
       clippedSchema: MessageType): Table = {
+    convertToUnsigned(evolveSchemaIfNeededAndCloseImp(inputTable, filePath, clippedSchema))
```

Comment:
Is it ok to alwalys do this casting for unsigned integrals ? Seems this is only needed from Spark 3.2.  What's the behavior for Spark version before 3.2 ?
Maybe it needs shims for this.

Reply:
No need Shims I think.
This PR is conformed to the CPU version behaviors of both Spark 3.2.0+ and Spark 3.2.0-.

The expected read type is decided by the parquet code in spark.
When reading unsigned integer,  Spark 3.2.0+ CPU version returns signed type and this PR does the same thing by casting.
And For Spark 3.2.0- just throws an exception for both CPU and GPU in spark parquet code.